java-client/src/main/java/com/wavefront/integrations/WavefrontDirectSender.java

```diff
@@ -70,14 +80,54 @@ public void send(String name, double value, @Nullable Long timestamp, String sou
     addPoint(name, value, timestamp, source, pointTags);
   }
 
+  @Override
+  public void send(HistogramGranularity histogramGranularity, List<Pair<Double, Integer>> distribution, String name) throws IOException {
```

Comment:
Just noticed, the javadoc for several send methods say we will use the machine hostname if source is null, but we use DEFAULT_SOURCE. Perhaps we can cache and reuse the hostname, something like this:

```
private String getDefaultSource() throws UnknownHostException {
    if (defaultSource == null) {
      defaultSource = InetAddress.getLocalHost().getHostName();
    }
    return defaultSource;
  }
```